app/lib/features/chat/providers/notifiers/chat_room_notifier.dart

```diff
@@ -38,7 +38,12 @@ class ChatRoomNotifier extends StateNotifier<ChatRoomState> {
     subscription = timeline?.diffRx().listen((event) async {
       await _parseEvent(event);
     });
-    await timeline?.paginateBackwards(10);
+    bool hasMore = false;
+    do {
+      hasMore = await timeline!.paginateBackwards(10);
+      // wait for diffRx to be finished
+      sleep(const Duration(milliseconds: 100));
```

Comment:
I have two problems with that:
1. sleep isn't `async` and thus _blocks_ the thread ... just want to avoid janking in the UI
2. 100ms might not be enough, is it? We can't guarantee that on slower phones ...

I wonder if there isn't a way for the `paginateBackwards(10)` to give us both information directly, rather than us relying on other channels hopefully having processed thing properly...

Reply:
We could add Future delay here. Current problem with pagination is that it gives 10 events which are not specifically message events (in all cases) and it doesn't add up enough messages in chat list to trigger pagination from UI. So this check is needed to ensure messages list are populated to determine if room pagination is possible or not. This is how the mechanism of ```flutter_chat_ui``` is set up.

Reply:
yeah, just keep it as is for now...

Reply:
Without this loop call, the UI wouldn't populate enough messages to reach the ```scrollExtent``` of whole list interface, which would allow scroll up gesture.